src/allmydata/storage_client.py

```diff
@@ -752,34 +782,56 @@ class AnnouncementNotMatched(Exception):
     """
 
 
+@attr.s(auto_exc=True)
```

Comment:
`auto_exc` has what I consider a significant misfeature so I've always explicitly turned it off when defining exceptions: it makes the instances compare by id instead of by value.  eg, with this definition, `MissingPlugin(foo) != MissingPlugin(foo)`.

Reply:
Seems like I could go either way on whether that's good or bad ;) but do you mean you can turn off _just_ that part, or do you mean you just never use `auto_exc=`?